evil-owl.el

```diff
@@ -213,22 +217,32 @@ MARK points nowhere."
                            ((markerp pos) (marker-buffer pos)))))
     (with-current-buffer buffer
       (let ((line (line-number-at-pos pos))
-            (column (save-excursion
-                      (goto-char pos)
-                      (current-column))))
-        (list line column (current-buffer))))))
+            file-text
+            column)
+        (save-excursion
+          (goto-char pos)
+          (setq column (current-column))
+          (let* ((sub-beg (line-beginning-position))
+                 (sub-end (if evil-owl-text-limit
+                              (min (+ sub-beg
+                                      evil-owl-text-limit)
+                                   (line-end-position))
+                            (line-end-position))))
+            (setq file-text (buffer-substring sub-beg sub-end))))
```

Comment:
If you move the `save-excursion` and `goto-char` outside of the initial `let`, you can have all the variables bound inside the `let` (probably changed to `let*`) without using `setq`.

Also, I think `context`, `context-beg`, and `context-end` might be better names for the new locals.